### PR TITLE
Aldaine

### PR DESCRIPTION
**Pull request by aldaineclarke**
Completed resource for the user resource

### CODE REVIEW

**Comment by aldaineclarke** [comment]
End Points to check are:
prefix = "api/v1"
/accounts full crud
/users full crud
/authenticate

File: controllers/user.controller.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+        /**
+     * 
+     * ### Description
+     * Gets all users that have created a profile in the database.
+     * @param {Request} req 
+     * @param {Response} res 
+     * @param {Next} next 
+     */
+    static getAllUsers = async (req, res, next)=>{
```
Comment (Dawnz) [inline controllers/user.controller.js]: an empty user list shold show a 200 (Ok status code) and return an empty array

File: controllers/user.controller.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+     * @param {Request} req 
+     * @param {Response} res 
+     * @param {Next} next 
+     */
+    static deleteUserProfile = async(req, res, next)=>{
+        try{
+            let id = req.params.id;
+            if(!ObjectId.isValid(id)) throw new Error("ID does not match any user profile in database");
+            let user = User.findByIdAndDelete(id);
+            JSONResponse.success(res, "Successfully deleted user", user, 203);
```
Comment (Dawnz) [inline controllers/user.controller.js]: a successful delete should give a 200 error code

File: controllers/user.controller.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        }catch(error){
+            JSONResponse.error(res, "Unable to delete user",error, 500 );
```
Comment (Dawnz) [inline controllers/user.controller.js]: avoid 500 error codes. this indicates that the server has crashed. 
if the user is not found use 404.

File: controllers/user.controller.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+     */
+    static getUserProfile = async(req, res, next)=>{
+        try{
+            let id = req.params.id;
+            if(!ObjectId.isValid(id)) throw new Error("Id is not a valid user profile in database");
+            let user = User.findById(id);
+            JSONResponse.success(res, "Retrieved user info", user, 200);
+
+        }catch(error){
+            JSONResponse.error(res, "Unable to find user", error, 500);
```
Comment (Dawnz) [inline controllers/user.controller.js]: same here

**Review by Dawnz — CHANGES_REQUESTED** [review]
please see review in user controller and make changes to other controllers

File: controllers/user.controller.js (hunk 2 above)
Comment (aldaineclarke) [reply controllers/user.controller.js]: added null check for all calls for single *item from database

**Review by aldaineclarke — COMMENTED** [review]
fixed all issues so far

**Comment by Dawnz** [comment]
@rdwayne876  and @Geo-the-Pitbull  The following endpoints can now be documented
-authenticate
-user
still testing transaction and accounts

**Pull request merged**